cms/static/js/spec/views/pages/course_outline_spec.js

```diff
@@ -872,61 +876,71 @@ define(["jquery", "common/js/spec_helpers/ajax_helpers", "common/js/components/u
                     expect($("#due_date").val()).toBe('7/10/2014');
                     expect($("#grading_type").val()).toBe('Lab');
                     expect($("#staff_lock").is(":checked")).toBe(true);
-                    expect($("#id_timed_exam").is(":checked")).toBe(false);
-                    expect($("#id_proctored_exam").is(":checked")).toBe(true);
+                    expect($("#id_timed_exam").is(":checked")).toBe(true);
+                    expect($("#id_proctored_exam").is(":checked")).toBe(false);
                     expect($("#id_not_timed").is(":checked")).toBe(false);
                     expect($("#id_practice_exam").is(":checked")).toBe(false);
                     expect($("#id_time_limit").val()).toBe("02:30");
+                    expect($("#id_hide_exam_after_due_date").is(":checked")).toBe(true);
                 });
 
-                it('can hide the time limit field when the None radio box is selected', function() {
+                it('can hide time limit and hide after due fields when the None radio box is selected', function() {
                     createCourseOutlinePage(this, mockCourseJSON, false);
                     outlinePage.$('.outline-subsection .configure-button').click();
+                    selectAdvancedSettings();
                     setEditModalValues("7/9/2014", "7/10/2014", "Lab", true);
                     selectDisableSpecialExams();
 
                     // id_time_limit_div should be hidden when None is specified
-                    expect($('#id_time_limit_div')).toHaveClass('is-hidden');
+                    expect($('#id_time_limit_div').is(':hidden')).toBe(true);
+                    expect($('.timed-exam-options-list-fields').is(':hidden')).toBe(true);
                 });
 
                 it('can select the practice exam', function() {
                     createCourseOutlinePage(this, mockCourseJSON, false);
                     outlinePage.$('.outline-subsection .configure-button').click();
+                    selectAdvancedSettings();
                     setEditModalValues("7/9/2014", "7/10/2014", "Lab", true);
                     selectPracticeExam("00:30");
                     // id_time_limit_div should not be hidden when practice exam is specified
-                    expect($('#id_time_limit_div')).not.toHaveClass('is-hidden"');
```

Comment:
This line (and others like it) were caused me a lot of grief. Notice the extraneous trailing `"` character? Nothing ever matches that class! This test had been verifying that certain elements did not contain a fictitious class, which was always true.

In investigating what was actually going on here, I learned 2 major things:
1. The time limit, review rules, and hide-after-due options are inside of divs that have an `is-hidden` class. This is always true, even when the divs are visible! We control visibility via `show()` and `hide()`, which set the element's `display` to be `none` and `block`, respectively. I tried removing the `is-hidden` class from the elements in the template, but that broke the modal's responsiveness so I let it be for now.
2. Many of these tests needed to be calling `selectAdvancedSettings()` in order to actually open the advanced settings tab, and confirm that certain elements were visible that should have been visible.